### ISSUE TITLE
FzfLua  Buffers: Arrow keys not working? Can't move around the buffers.

### ISSUE THREAD

**Issue by vinoff** [ibhagwan/fzf-lua#887]
I am looking for an extension that allows be to pick buffers and guess what, Fzf-lua, which I am already using, seems to do that as well! Sadly, I can't seem to be able to use the arrow keys while doing `FzfLua buffers`.. Any idea what's happening?

```
Fzf-lua: latest version.

NVIM v0.9.2
Build type: Release
LuaJIT 2.1.1694285958
```

EDIT: I should say that the arrow keys (up/down) work everyone else. files, lines, etc..

**Comment by ibhagwan** [comment]
It should work by default, maybe you have a terminal mapping that overrides down/up?

Does ctrl-j|ctrl-k work? Have you tried with mini.sh?

**Comment by vinoff** [comment]
No, C-j and C-k do not work.

I have just tried with mini.sh by doing `sh -c "$(curl -s https://raw.githubusercontent.com/ibhagwan/fzf-lua/main/scripts/mini.sh)"`  as per the README and the arrow keys still don´t work! C-j and C-k also don´t work..

**Comment by ibhagwan** [comment]
Can you run `fzf` in the shell and see if it works there? What’s in your `echo $FZF_DEFAULT_OPTS`?

**Comment by vinoff** [comment]
Adding that to my config seems to make it work, though now it opens buffers as soon as I open vim. Here is my config btw:

```
return {
	"ibhagwan/fzf-lua",
	-- optional for icon support
	dependencies = { "nvim-tree/nvim-web-devicons" },
	config = function()
		-- calling `setup` is optional for customization
		require("fzf-lua").setup({
			winopts = {
				preview = { default = "bat_native" },
			},
			file_ignore_patterns = { "node_modules/.*", ".*%.jpg", ".*%.png", ".*%.gif", ".*%.jpeg" },
			-- grep = {
			--     rg_opts = "--smart-case -g '!{.git,node_modules}/*'",
			-- },
			grep = {
				rg_opts = "--sort-files --hidden --column --line-number --no-heading "
					.. "--color=always --smart-case -g '!{.git,node_modules}/*'",
			},
		})
		require("fzf-lua").buffers({ keymap = { fzf = { ["down"] = "down", ["up"] = "up" } } })
		require("fzf-lua").buffers({ keymap = { fzf = { ["ctrl-j"] = "down", ["ctrl-k"] = "up" } } })
		require("fzf-lua").buffers({ keymap = { fzf = { ["ctrl-n"] = "down", ["ctrl-p"] = "up" } } })
	end,
}

```

And keymaps:
```

vim.keymap.set("n", "<C-p>", "<cmd>lua require('fzf-lua').files()<CR>", { silent = true })
vim.keymap.set("n", "<c-f>", "<cmd>lua require('fzf-lua').live_grep_native()<CR>", { silent = true })
vim.keymap.set("n", "<c-a>", "<cmd>lua require('fzf-lua').buffers()<CR>", { silent = true })
```

**Comment by vinoff** [comment]
Here is the screenshot you requested:

![fzf](https://github.com/ibhagwan/fzf-lua/assets/90382765/6debd55d-b706-45dd-9b8c-5c08799404d2)

**Comment by ibhagwan** [comment]
> Adding that to my config seems to make it work, though now it opens buffers as soon as I open vim. Here is my config btw

These lines were not meant to be added to the config, just to test via `:lua …`.

If you wish to add these to your setup do:
```lua
require("fzf-lua").setup({
   … your other setup options …
   keymap = {
     fzf = {
      ['down']       = 'down',
      ['up']         = 'up',
      ["ctrl-z"]     = "abort",
      ["ctrl-u"]     = "unix-line-discard",
      ["ctrl-f"]     = "half-page-down",
      ["ctrl-b"]     = "half-page-up",
      ["ctrl-a"]     = "beginning-of-line",
      ["ctrl-e"]     = "end-of-line",
      ["alt-a"]      = "toggle-all",
      ["f3"]         = "toggle-preview-wrap",
      ["f4"]         = "toggle-preview",
      ["shift-down"] = "preview-page-down",
      ["shift-up"]   = "preview-page-up",
   },
}
```

Alternatively just add these options to the mapping:
```lua

vim.keymap.set("n", "<c-a>", "<cmd>lua require('fzf-lua').buffers({ keymap = { fzf = { ['down']='down',   ['up']='up' } } })<CR>", { silent = true })
```

**Comment by ibhagwan** [comment]
> Here is the screenshot you requested:

That’s not the one, I requested you press F1 when buffers is open so I can see the fzf-lua keybinds.

**Comment by vinoff** [comment]
> > Here is the screenshot you requested:
> 
> That’s not the one, I requested you press F1 when buffers is open so I can see the fzf-lua keybinds.

Sorry. In my defense, I am an idiot. Here is the requested screenshot:


![fzf2](https://github.com/ibhagwan/fzf-lua/assets/90382765/7a2710e5-2c6f-439f-80e2-07642c6c904b)

**Comment by vinoff** [comment]
Adding this to my mappings did not help:

`vim.keymap.set("n", "<c-a>", "<cmd>lua require('fzf-lua').buffers({ keymap = { fzf = { ['down']='down',   ['up']='up' } } })<CR>", { silent = true })`

**Comment by vinoff** [comment]
Sorry, but with the config I posted above (with the one keymaps you just posted) it seems to be working.. I dont know why it didnt work before.

In sum, this works:

return {
	"ibhagwan/fzf-lua",
	-- optional for icon support
	dependencies = { "nvim-tree/nvim-web-devicons" },
	config = function()
		-- calling `setup` is optional for customization
		require("fzf-lua").setup({
			winopts = {
				preview = { default = "bat_native" },
			},
			file_ignore_patterns = { "node_modules/.*", ".*%.jpg", ".*%.png", ".*%.gif", ".*%.jpeg" },
			-- grep = {
			--     rg_opts = "--smart-case -g '!{.git,node_modules}/*'",
			-- },
			grep = {
				rg_opts = "--sort-files --hidden --column --line-number --no-heading "
					.. "--color=always --smart-case -g '!{.git,node_modules}/*'",
			},
			keymap = {
				fzf = {
					["down"] = "down",
					["up"] = "up",
					["ctrl-z"] = "abort",
					["ctrl-u"] = "unix-line-discard",
					["ctrl-f"] = "half-page-down",
					["ctrl-b"] = "half-page-up",
					["ctrl-a"] = "beginning-of-line",
					["ctrl-e"] = "end-of-line",
					["alt-a"] = "toggle-all",
					["f3"] = "toggle-preview-wrap",
					["f4"] = "toggle-preview",
					["shift-down"] = "preview-page-down",
					["shift-up"] = "preview-page-up",
				},
			},
		})
	end,
}

**Comment by ibhagwan** [comment]
> Sorry, but with the config I posted above (with the one keymaps you just posted) it seems to be working.. I dont know why it didnt work before.
> In sum, this works:

That's great! not sure what's overriding your defaults, but as long as this works we can close this issue.

**Comment by vinoff** [comment]
Thank you @ibhagwan!

**Comment by vinoff** [comment]
Shit. I found out what the mistake was. Human error. I am an absolute idiot. I am sorry @ibhagwan.

This was the mistake:

I was testing things with 2 buffers opened. With 2 buffers opened, even though you see 2 buffers in the panel, you cannot actually select both. You cannot move/arrow between them. To properly test this out, I have to have at least 3 buffers opened. At some point during the whole fiasco I started opening more buffers, and that is why it started working.
